app/services/events/create_service.rb

```diff
@@ -49,7 +49,7 @@ def produce_kafka_event(event)
           external_customer_id: event.external_customer_id,
           external_subscription_id: event.external_subscription_id,
           transaction_id: event.transaction_id,
-          timestamp: event.timestamp,
+          timestamp: event.timestamp.to_i,
```

Comment:
It might be an issue for the weighted sum aggregation as we need to handle events with millisecond granularity

I think you could use `to_f`, but we have to check if the kafka topic handle it correctly

Reply:
updated!